contracts/DEX.sol

```diff
@@ -24,13 +26,19 @@ contract DEX {
         uint256 ssvAmount = amount / rate;
         cdtToken.transferFrom(msg.sender, address(this), amount);
         ssvToken.transfer(msg.sender, ssvAmount);
+        cdtAllowance[msg.sender] += amount;
         emit CDTToSSVConverted(ssvAmount);
     }
 
     function convertSSVToCDT(uint256 amount) public {
         uint256 cdtAmount = amount * rate;
+        require(
+            cdtAllowance[msg.sender] >= cdtAmount,
+            "You can't get back amount of CDT tokens more than exchanged before"
```

Comment:
I would use a more general message (not using the word "You").

Reply:
fixed